src/mixins/nft.js

```diff
@@ -514,14 +514,15 @@ export default {
           this.classId,
           1
         );
-        const txHash = await broadcastTx(signData, this.getSigner);
+        const { txHash, code } = await broadcastTx(signData, this.getSigner);
         logTrackerEvent(
           this,
           'NFT',
           'NFTCollectBroadcastTxComplete',
           this.classId,
           1
         );
+        if (code) throw new Error(`TX_FAILED_WITH_CODE_${code}`);
```

Comment:
I think we should explicitly check for `0`
```suggestion
        if (code !== 0) throw new Error(`TX_FAILED_WITH_CODE_${code}`);
```